### ISSUE TITLE
Regression for "= ANY(?::int[])" no longer produces array bind variable

### ISSUE THREAD

**Issue by paplorinc** [jOOQ/jOOQ#9224]
### Expected behavior and actual behavior:

Probably related to https://github.com/jOOQ/jOOQ/commit/af024cde2d4966edcf453999f61db43e68f3e651#diff-542330b86eb85001a6f33606701bdf91R118
Expressions are verbosely spelled out as disjunctive terms, instead of as a list - as it was previously.

### Steps to reproduce the problem (if possible, create an MCVE: https://github.com/jOOQ/jOOQ-mcve):

In `3.11.11`
```java
db.selectOne()
    .where(DSL.inline(0).eq(DSL.any(0, 0)))
```
will generate (edited):
```sql
SELECT 1 AS "one"
WHERE 0 = ANY (
 CAST('{"0","0"}' AS int[])
)
```
but in `3.12.1` it's:
```sql
SELECT 1 AS "one"
WHERE (
 0 = 0
 OR 0 = 1
)
```
For larger queries this become unexecutable.

### Versions:

- jOOQ: 3.12.1
- Java: 11.0.4
- Database (include vendor): Postgres 11.4
- OS: Mac
- JDBC Driver (include name if inofficial driver): 42.2.6

### Workaround
Postgres explain seems to indicate that the previous behavior can also be achieved via an in-predicate, like:
```java
db.selectOne()
    .where(DSL.inline(0).in(0, 0))
```
which generates in both JOOQ versions:
```sql
SELECT 1 AS "one"
WHERE 0 IN (
 0, 0
)
```
Is there any downside to using the latter? (parsing speed, [JDBC parameter limit](https://stackoverflow.com/a/42251312) in Postgres or anything that `= ANY` doesn't have?)

**Comment by lukaseder** [comment]
Thank you very much for your report. Indeed, that's a significant performance regression, which went unnoticed because our integration tests still worked - the alternative SQL using `or` predicates is correct, but no longer profits from the array bind variable.

I also think this was introduced with the `like(QuantifiedSelect)` predicate.

We'll look into this immediately and backport a fix to 3.12.2 (#9228)

**Comment by lukaseder** [comment]
Notice, I cannot reproduce a cast to `any[]` in version 3.11.11. I'm getting `int[]` with your code. How did you generate that `any[]` array type?

**Comment by paplorinc** [comment]
> How did you generate that any[] array type?

With version `3.11.11`:
<img src="https://user-images.githubusercontent.com/1841944/65045229-b9de0800-d95e-11e9-94c1-f0c8b69d6133.png">

Would you like me to send you our context settings?

**Comment by lukaseder** [comment]
Interesting - it looks like another bug, but I cannot reproduce it:

```java
public static void main(String[] args) {
    System.out.println(
        DSL.using(SQLDialect.POSTGRES)
            .selectOne()
            .where(DSL.inline(0).eq(DSL.any(0, 0)))
    );
}
```

Yields:

```sql
select 1 as "one"
where 0 = any (
  cast('{"0","0"}' as int[])
)
```

What does your `db` object look like?

**Comment by paplorinc** [comment]
Interesting, printing it out from `main` results in an `int[]`, but from the Idea's Evaluate Expression window it's `any[]`
<img src="https://user-images.githubusercontent.com/1841944/65054409-ca49af00-d96d-11e9-826b-c7191e2a6ae2.png">
Same from the Groovy console.
The JShell console is also correct again, so something's probably lost among some Groovy magic.

**Comment by lukaseder** [comment]
OK, that settles that :) But we'll definitely fix the other, main regression ASAP

**Comment by lukaseder** [comment]
Fixed for 3.13.0. Backport scheduled for 3.12.2 (#9228)
